forge-gui/res/cardsfolder/m/magar_of_the_magic_strings.txt

```diff
@@ -2,10 +2,13 @@ Name:Magar of the Magic Strings
 ManaCost:1 B R
 Types:Legendary Creature Minotaur Performer
 PT:3/3
-A:AB$ ChangeZone | Cost$ 1 B R | ValidTgts$ Instant.YouOwn,Sorcery.YouOwn | TgtPrompt$ Select target instant or sorcery card in your graveyard | Origin$ Graveyard | Destination$ Battlefield | RememberLKI$ True | FaceDown$ True | FaceDownPower$ 3 | FaceDownToughness$ 3 | FaceDownSetType$ Creature | AnimateSubAbility$ DBAnimate | StackDescription$ REP Note_{p:You} notes & target instant or sorcery card in your graveyard_{c:Targeted} & put_puts | SpellDescription$ Note the name of target instant or sorcery card in your graveyard and put it onto the battlefield face down. It's a 3/3 creature with "Whenever this creature deals combat damage to a player, you may create a copy of the card with the noted name. You may cast the copy without paying its mana cost" and "If this creature would leave the battlefield, exile it instead of putting it anywhere else."
-SVar:DBAnimate:DB$ Animate | RememberObjects$ RememberedLKI | Defined$ Remembered | Triggers$ DealsCDTrig | Replacements$ ReplaceDies | Duration$ Permanent
+A:AB$ Pump | Cost$ 1 B R | ValidTgts$ Instant.YouOwn,Sorcery.YouOwn | TgtZone$ Graveyard | TgtPrompt$ Select target instant or sorcery card in your graveyard | SubAbility$ DBNoteName | StackDescription$ REP Note_{p:You} notes & target instant or sorcery card in your graveyard_{c:Targeted} & put_puts | SpellDescription$ Note the name of target instant or sorcery card in your graveyard and put it onto the battlefield face down. It's a 3/3 creature with "Whenever this creature deals combat damage to a player, you may create a copy of the card with the noted name. You may cast the copy without paying its mana cost" and "If this creature would leave the battlefield, exile it instead of putting it anywhere else."
+SVar:DBNoteName:DB$ NameCard | ChooseFromDefinedCards$ Targeted | SubAbility$ DBChangeZone | StackDescription$ None
+SVar:DBChangeZone:DB$ ChangeZone | Defined$ Targeted | Origin$ Graveyard | Destination$ Battlefield | FaceDown$ True | FaceDownPower$ 3 | FaceDownToughness$ 3 | FaceDownSetType$ Creature | AnimateSubAbility$ DBAnimate | SubAbility$ DBCleanup | StackDescription$ None
+SVar:DBAnimate:DB$ Animate | NoteNamedCard$ True | Defined$ Remembered | Triggers$ DealsCDTrig | Replacements$ ReplaceDies | Duration$ Permanent
 SVar:DealsCDTrig:Mode$ DamageDone | ValidSource$ Card.Self | ValidTarget$ Player | CombatDamage$ True | Execute$ TrigPlay | TriggerZones$ Battlefield | TriggerDescription$ Whenever this creature deals combat damage to a player, you may create a copy of the card with the noted name. You may cast the copy without paying its mana cost.
-SVar:TrigPlay:DB$ Play | Defined$ RememberedLKI | ZoneRegardless$ True | CopyCard$ True | ValidSA$ Spell | WithoutManaCost$ True | Optional$ True
+SVar:TrigPlay:DB$ Play | CopyFromChosenName$ True | ValidSA$ Spell | WithoutManaCost$ True | Optional$ True
```

Comment:
uuhh, you may want to explain what you determined as the cause of the crash so I can understand :)
this wouldn't really be a correct way for fix, I think I pointed out it's supposed to work like this back when you implemented it:
> 707.14. One card (Magar of the Magic Strings) instructs a player to note the name of a particular card in a graveyard and create a copy of the card with the noted name. To do so, use the characteristics of that card as it last existed in the graveyard to determine the copiable values of the copy.

Reply:
Ah I forgot about that part of the CR and just re read the card
Sorry it's been so long since I looked at this that I forgot some of the prior conversation 😅 

I think the cause of the crash is in the second collectSpellsForPlayEffect call in getSpellsFromPlayEffect is sending state as null

I'll debug it some more

Reply:
OK on further digging it seems that the copied card made by PlayEffect isModal (for AbilityUtils.getSpellsFromPlayEffect) but it doesn't have the Modal state

Reply:
Hmm this also seems wrong... and comes a bit earlier
<img width="795" alt="image" src="https://github.com/user-attachments/assets/62831667-0681-4aac-a70a-6ef7643bf674">